internal/server/server.go

```diff
@@ -90,40 +133,21 @@ func (s *Server) UpdateUserStatus() {
 	}
 }
 
-func (s *Server) AddClient(client i.ClientInterface) {
-	s.Clients = append(s.Clients, client)
-}
-
-func (s *Server) SendServerPrivateMessage(message string, client i.ClientInterface) {
-	services.PrivateServerMessage(message, client)
-}
-
-func readMessage(conn net.Conn) (string, error) {
-	reader := bufio.NewReader(conn)
-	message, err := reader.ReadString('\n')
-
-	if err != nil {
-		return "", err
-	}
-	return strings.TrimSpace(message), nil
-}
-
 func (s *Server) HandleConn(conn net.Conn) {
 	defer conn.Close()
 
 	ipAddr := conn.RemoteAddr().String()
 
 	for {
-		message, err := readMessage(conn)
+		message, err := utils.ReadMessage(conn)
 		if err != nil {
 			return
 		}
 
 		client := s.getClientByIP(ipAddr)
 		if client == nil {
 			client = cl.NewClient(conn, ipAddr)
-			s.SendServerPrivateMessage(fmt.Sprintf("%s connected.", client.GetUsername()), client)
-			s.CommandHandler.Handle(client, "/join")
+			s.SendServerPrivateMessage(fmt.Sprintf("A new user connected. "), client)
```

Comment:
do something like pliz join and login ?

Reply:
I like your concept but, we've a problem this runs before actual handler and this presents an interesting issue of UX. I've tried it but looking for better solutions.